api/app/models.py

```diff
@@ -237,17 +237,25 @@ def __init__(self, *args, **kwargs) -> None:
     def vuln_matching_ecosystem(self) -> str:
         """
         Returns the ecosystem string for vulnerability matching.
-        If the ecosystem starts with "alpine-",
-        it change the value to include only the minor version.
-        Example: "alpine-3.22.0" → "alpine-3.22"
+        If the ecosystem starts with "alpine-", it keeps the major and minor version.
+        If it starts with "rocky-", it keeps only the major version.
+        Example: "alpine-3.22.0" → "alpine-3.22", "rocky-9.3" → "rocky-9"
         For other ecosystems, returns the original value.
         """
+        if not self.ecosystem:
+            return self.ecosystem
         if self.ecosystem and self.ecosystem.startswith("alpine-"):
```

Comment:
```suggestion
        if self.ecosystem.startswith("alpine-"):
```
このステップではecosystemの空チェック不要になった